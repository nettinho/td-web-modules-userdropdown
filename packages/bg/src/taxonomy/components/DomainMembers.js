import _ from "lodash/fp";
import React, { Fragment } from "react";
import PropTypes from "prop-types";
import { Card, Message, Grid } from "semantic-ui-react";
import { Link } from "react-router-dom";
import { compose } from "redux";
import { connect } from "react-redux";
import { injectIntl, FormattedMessage } from "react-intl";
import { Loading } from "@truedat/core/components";
import { linkTo } from "../routes";
import DomainMember from "./DomainMember";
import DomainMembersActions from "./DomainMembersActions";
const toAddMemberPath = linkTo.DOMAIN_MEMBERS_NEW;

export const DomainMembers = ({
  domain,
  domainMembers,
  domainMembersLoading,
  intl: { formatMessage }
}) => {
  const available_actions = {
    create_or_update: {
      order: 3,
      icon: "add user",
      messageId: "domain.actions.add_member",
      action: id => toAddMemberPath({ id })
    }
  };
  return (
    <Fragment>
      {!_.isEmpty(domainMembers) &&
        !domainMembersLoading && (
          <Grid centered columns={1}>
            <Grid.Column>
              <DomainMembersActions availableActions={available_actions} />
            </Grid.Column>
          </Grid>
        )}
      {domainMembersLoading && <Loading inline="centered" />}
      {_.isEmpty(domainMembers) &&
        !domainMembersLoading && (
          <Message
            icon="warning"
            header={formatMessage({ id: "domain.members.empty" })}
            list={[
              <Link key={1} to={linkTo.DOMAIN_MEMBERS_NEW({ id: domain.id })}>
                <FormattedMessage id="domain.actions.add_member" />
              </Link>
            ]}
          />
        )}
      <Card.Group>
        {domainMembers.map((m, i) => <DomainMember key={i} {...m} />)}
      </Card.Group>
    </Fragment>
  );
};

DomainMembers.propTypes = {
  domain: PropTypes.object,
  domainMembers: PropTypes.array,
  domainMembersLoading: PropTypes.bool,
  intl: PropTypes.object
};

const mapStateToProps = ({ domain, domainMembers, domainMembersLoading }) => ({
  domain,
  domainMembers,
  domainMembersLoading
});

export default compose(
  injectIntl,
  connect(mapStateToProps)
)(DomainMembers);
