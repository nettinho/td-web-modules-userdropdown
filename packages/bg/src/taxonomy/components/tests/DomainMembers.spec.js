import React from "react";
import { shallowWithIntl } from "intl-stub";
import { DomainMembers } from "../DomainMembers";

describe("<DomainMembers />", () => {
  const domainMembersLoading = false;
  const domainMembers = [{ id: 1, name: "d1" }, { id: 2, name: "d2" }];
  const props = { domainMembers, domainMembersLoading };

  it("matches the latest snapshot", () => {
    const wrapper = shallowWithIntl(<DomainMembers {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
