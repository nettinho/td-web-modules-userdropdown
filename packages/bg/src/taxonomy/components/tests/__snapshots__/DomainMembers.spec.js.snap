// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DomainMembers /> matches the latest snapshot 1`] = `
<React.Fragment>
  <Grid
    centered={true}
    columns={1}
  >
    <GridColumn>
      <Connect(DomainMembersActions)
        availableActions={
          Object {
            "create_or_update": Object {
              "action": [Function],
              "icon": "add user",
              "messageId": "domain.actions.add_member",
              "order": 3,
            },
          }
        }
      />
    </GridColumn>
  </Grid>
  <CardGroup>
    <InjectIntl(Connect(DomainMember))
      id={1}
      key="0"
      name="d1"
    />
    <InjectIntl(Connect(DomainMember))
      id={2}
      key="1"
      name="d2"
    />
  </CardGroup>
</React.Fragment>
`;
